gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthorizationService.js

```diff
@@ -100,125 +100,510 @@ var fluid = fluid || require("infusion");
+    gpii.oauth2.authorizationService.doGrant = function (dataStore, codeGenerator, authDecisionInfo) {
         // Generate the authorization code and record it
         var code = codeGenerator.generateAuthCode();
-        dataStore.saveAuthCode(authDecision.id, code);
-        return code;
+        var savePromise = dataStore.saveAuthCode(authDecisionInfo.id, code);
+        var mapper = function () {
+            return code;
+        };
+        return fluid.promise.map(savePromise, mapper);
+    };
+    // ==== End of grantAuthorizationCode()
+
+    // ==== addAuthorization()
+    gpii.oauth2.authorizationService.addAuthorization = function (that, gpiiToken, oauth2ClientId, selectedPreferences) {
+        var input = {
+            gpiiToken: gpiiToken,
+            oauth2ClientId: oauth2ClientId,
+            selectedPreferences: selectedPreferences
+        };
+        return fluid.promise.fireTransformEvent(that.events.onAddAuthorization, input);
+    };
+
+    // Shared by getSelectedPreferences() and setSelectedPreferences()
+    // @dataStore (Object): a data store instance
+    // @input (Object): Acceps this structure:
+    // {
+    //     gpiiToken: (String),
+    //     oauth2ClientId: (String),
+    //     selectedPreferences: (Object)
+    // }
+    // @return (Promise): a promise object to be passed to the next processing function
+    gpii.oauth2.authorizationService.findGpiiToken = function (dataStore, input) {
+        var gpiiTokenPromise = dataStore.findGpiiToken(input.gpiiToken);
+        var mapper = function (gpiiToken) {
+            return {
+                gpiiToken: gpiiToken,
+                inputArgs: input
+            };
+        };
+        return fluid.promise.map(gpiiTokenPromise, mapper);
+    };
+
+    // @dataStore (Object): a data store instance
+    // @record (Object): Acceps this structure:
+    // {
+    //     gpiiToken: {Object},   // A GPII token record
+    //     inputArgs: {
+    //         gpiiToken: (String),
+    //         oauth2ClientId: (String),
+    //         selectedPreferences: (Object)
+    //     }
+    // }
+    // @return (Promise): a promise object to be passed to the next processing function
+    gpii.oauth2.authorizationService.findClient = function (dataStore, record) {
+        var clientPromise = dataStore.findClientByOauth2ClientId(record.inputArgs.oauth2ClientId);
+        var mapper = function (client) {
+            return fluid.extend({}, record, {client: client});
+        };
+        return fluid.promise.map(clientPromise, mapper);
     };
 
-    gpii.oauth2.authorizationService.addAuthorization = function (dataStore, codeGenerator, userId, oauth2ClientId, selectedPreferences) {
-        var client = dataStore.findClientByOauth2ClientId(oauth2ClientId);
-        if (client) {
-            var clientId = client.id;
-            var redirectUri = client.redirectUri;
+    // @dataStore (Object): a data store instance
```

Comment:
The doc style has a comment beginning /**
and then reads @param dataStore {Object}

We don't have anything that parses these but we may as well be consistent in case oneday we do.